### PR TITLE
FOUNDATIONS : Hash Password

### LINKED ISSUE #361: FOUNDATIONS: Add `Password` Rules Validations in UserService.cs

### LINKED ISSUE #363: FOUNDATIONS: Hash Password in SecurityService.cs
After hashing make a code rub commit to use this function in signup method in UserSecurityOrchestration Service

### PR DESCRIPTION
**Pull request by ShohruhUzDev**
Closes #363 
Closes #361

### CODE REVIEW

File: Tarteeb.Api/Brokers/Tokens/TokenBroker.cs [hunk 1]
```diff
@@ -52,5 +53,14 @@ public string GenerateJWT(User user)
...
         }
+
+        public string HashToken(string password)
+        {
+            using (var sha256 = SHA256.Create())
+            {
+                var hashedBytes = sha256.ComputeHash(Encoding.UTF8.GetBytes(password));
+                return BitConverter.ToString(hashedBytes).Replace("-", "").ToLower();
+            }
+        }
```
Comment (ElbekDeveloper) [inline Tarteeb.Api/Brokers/Tokens/TokenBroker.cs:64]: You cannot implement broker in foundation service:
 public string HashToken(string password) =>
    throw new NotImplementedException();

ga o'zgartirib. Buning uchun alohida brokers PR oching

File: Tarteeb.Api/Services/Orchestrations/UserSecurityOrchestrationService.cs [hunk 2]
```diff
@@ -38,6 +38,7 @@ public partial class UserSecurityOrchestrationService : IUserSecurityOrchestrati
         public ValueTask<User> CreateUserAccountAsync(User user, string requestUrl) =>
         TryCatch(async () =>
         {
+            user.Password=this.securityService.HashPassword(user.Password);
```
Comment (ElbekDeveloper) [inline Tarteeb.Api/Services/Orchestrations/UserSecurityOrchestrationService.cs:41]: format around `=` sign

File: Tarteeb.Api.Tests.Unit/Tarteeb.Api.Tests.Unit.csproj [hunk 3]
```diff
@@ -13,6 +13,7 @@
 		<PackageReference Include="FluentAssertions.Json" Version="6.1.0" />
 		<PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.0.0" />
 		<PackageReference Include="Moq" Version="4.18.2" />
+		<PackageReference Include="PasswordGenerator" Version="2.1.0" />
```
Comment (ElbekDeveloper) [inline Tarteeb.Api.Tests.Unit/Tarteeb.Api.Tests.Unit.csproj:16]: Bu qanaqa kutubxona ekan. Bizga bu kerak emas.

Comment (ElbekDeveloper) [reply Tarteeb.Api.Tests.Unit/Tarteeb.Api.Tests.Unit.csproj:16]: Oddiy string bilan tuzib chiqing. Object filler bunga imkon berishi mumkin.

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Securities/SecurityServiceTests.Validations.Hash.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+               string invalidstring)
+        {
+            // given
+            string randomPassword = invalidstring;
+            string invalidPassword = randomPassword;
+            var invalidUserException = new InvalidUserException();
+
+            invalidUserException.AddData(
+                key: nameof(User.Password),
+                values: "Password is not valid");
```
Comment (ElbekDeveloper) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Securities/SecurityServiceTests.Validations.Hash.cs:67]: Bu yerda parolda nima muammoligi ko'rinmayaptiku

**Review by ElbekDeveloper — CHANGES_REQUESTED** [review]

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Securities/SecurityServiceTests.Validations.Hash.cs (hunk 4 above)
Comment (ShohruhUzDev) [reply Tarteeb.Api.Tests.Unit/Services/Foundations/Securities/SecurityServiceTests.Validations.Hash.cs:67]: Invalid Password kelayapdiku

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Securities/SecurityServiceTests.Validations.Hash.cs [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+            // given
+            string randomPassword = invalidstring;
+            string invalidPassword = randomPassword;
+            var invalidPasswordException = new InsecurePasswordException();
+
+            invalidPasswordException.AddData(
+                key: "LengthComplexity",
+                values: "At least 8 characters required");
+
+            invalidPasswordException.AddData(
```
Comment (ElbekDeveloper) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Securities/SecurityServiceTests.Validations.Hash.cs:84]: bu kerakmas

**Pull request closed without merging**